#!/usr/bin/env python

PACKAGE="navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       name,                type,   level, description,           default, min, max
gen.add("points_height",            double_t, 0,  "max relevant points height: ", 2,    0.3, 10.0)
gen.add("points_height_start",            double_t, 0,  "min relevant points height: ", -0.5,    -1.0, 10.0)
gen.add("lateral_dist",            double_t, 0,  "max lateral dist to consider on free side: ", 10.0,    0.0, 120.0)
gen.add("r",            double_t, 0,  "radius around robot: ", 3.0,    1.0, 120.0)
gen.add("downsampling_cube",            double_t, 0,  "cube dimension for downsampling: ", 0.05,    0.0001, 1.0)
gen.add("outliers_filtering", bool_t, 0, "Remove outliers:", False)
gen.add("meanK",            int_t, 0,  "checked neighbors in outliers removal: ", 50,    1, 10000)
gen.add("stddevMulThresh",            double_t, 0,  "stddev threshold in outliers removal: ", 1.0,    0.0001, 10.0)
exit(gen.generate(PACKAGE, "filtering_param", "FilteringParam"))
